Replace `/proc/self/fd/<fdn>` with `/proc/<pid>/fd/<fdn>` for Linux

## Code
``` go
func main() {
	exe, err := memexec.New([]byte("#!/bin/bash\necho 'hello'\n"))
	if err != nil {
		panic(err)
	}
	defer exe.Close()

	stderr, _ := cmd.StderrPipe()
	go io.Copy(os.Stderr, stderr)
	stdout, _ := cmd.StdoutPipe()
	go io.Copy(os.Stdout, stdout)

	err = cmd.Run()
	if err != nil {
		panic(err)
	}
}

```
Output: 
```
/bin/bash: /proc/self/fd/3: No such file or directory
panic: exit status 127

goroutine 1 [running]:
main.main()
	/home/lindsay/workdir/exe/main.go:26 +0x272
exit status 2
```

## Problem
When the command starts, according to shebang, it creates another `bash` process .   
For bash process, `/proc/self` is linked to its own proc directory rather than original golang process.  
Bash process can not find script file and exit with error.   

Replace [calling path](https://github.com/amenzhinsky/go-memexec/blob/9e9712f1350c2ced40ab8629ab6e6e5c8315e3c8/memexec_linux.go#L18) with `/proc/<pid>/fd/<fdn>`  maybe can avoid this problem.   

## Platform
Go 1.20.2-1
Archlinux 6.2.7-arch1-1
Bash 5.1.016-3

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -2,10 +2,10 @@ name: Test and Lint
 on:
   push:
     branches:
-      - master
+      - main
   pull_request:
     branches:
-      - master
+      - main
 
 jobs:
   test:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -21,7 +21,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.19
       - name: Run Tests
         run: go test -v ./...
 
@@ -32,9 +32,9 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.19
       - name: Run golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.48.0
+          version: v1.52.0
           args: --enable=gofumpt,goimports,whitespace,gocritic,exportloopref,unconvert,prealloc

--- a/go.mod
+++ b/go.mod
@@ -1,5 +1,5 @@
 module github.com/amenzhinsky/go-memexec
 
-go 1.17
+go 1.19
 
 require golang.org/x/sys v0.0.0-20210903071746-97244b99971b

--- a/memexec_linux.go
+++ b/memexec_linux.go
@@ -15,7 +15,7 @@ func open(b []byte) (*os.File, error) {
 	if err != nil {
 		return nil, err
 	}
-	f := os.NewFile(uintptr(fd), fmt.Sprintf("/proc/self/fd/%d", fd))
+	f := os.NewFile(uintptr(fd), fmt.Sprintf("/proc/%d/fd/%d", os.Getpid(), fd))
 	if _, err := f.Write(b); err != nil {
 		_ = f.Close()
 		return nil, err

--- a/memexec_test.go
+++ b/memexec_test.go
@@ -1,6 +1,7 @@
 package memexec
 
 import (
+	"bytes"
 	"context"
 	"os"
 	"os/exec"
@@ -21,6 +22,26 @@ func TestCommand(t *testing.T) {
 	c := exe.Command("test")
 	if have := runCommand(t, c); !strings.HasPrefix(have, "test") {
 		t.Errorf("command output = %q doesn't contain %q", have, "test")
+	}
+}
+
+func TestCommand_Shell(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	exe, err := New([]byte("#!/bin/sh\necho hello"))
+	if err != nil {
+		panic(err)
+	}
+	defer exe.Close()
+
+	b, err := exe.Command().CombinedOutput()
+	if err != nil {
+		t.Fatalf("err = %v, output = %q", err, b)
+	}
+	if !bytes.Equal(b, []byte("hello\n")) {
+		t.Fatalf("output = %q, want %q", string(b), []byte("hello\n"))
 	}
 }
 
@@ -104,7 +125,7 @@ func runCommand(t *testing.T, cmd *exec.Cmd) string {
 	b, err := cmd.CombinedOutput()
 	if err != nil {
 		if b != nil {
-			t.Fatal(string(b))
+			t.Fatalf("failed to run: %q", string(b))
 		}
 		t.Fatal(err)
 	}

